File: src/WebUI/src/app/client/CustomerDto.ts

export interface ICustomer {
  customerID: number;
  name: string | undefined;
  pAN: string | undefined;
  addressPremises: string | undefined;
  adressLine1: string | undefined;
  addressLine2: string | undefined;
  city: string | undefined;
  pinCode: string | undefined;
  mobileNo: string | undefined;
  emailID: string | undefined;
  dateOfBirth: Date;
  stateId: number;
  isTracesRegistered: boolean;
  tracesPassword: string | undefined;
  tracesLogin: string | undefined;
  allowForm16B: boolean;
  customerProperty?: ICustomerProperty[] | undefined;
}

export interface ICustomerProperty {
  customerPropertyId: number;
  customerShare: number;
  customerId: number;
  propertyId: number;
  dateOfSubmission: Date;
  unitNo: string;
  remarks: string | undefined;
  isShared: boolean;
  statusTypeId: number;
  paymentMethodId: number;
  gstrate: number;
  tdsrate: number;
  totalUnitCost: number;
  agreementDate: Date; 
  tdsCollectedBySeller: boolean;
  dateOfAgreement: Date;
}


export interface ICustomerShareList {
  customerID?: number;
  name: string | undefined;
  share: string | undefined;
  form16b?: number ;
}

export interface ICustomerVM {
  customers ?: ICustomer [] | undefined;
  customerProperty?: ICustomerProperty[] | undefined;
}

File: src/Application/Customers/CustomerDto.cs
using AutoMapper;
using ReProServices.Application.Common.Mappings;
using System;
using System.Collections.Generic;

namespace ReProServices.Application.Customers
{
    public class CustomerDto : IMapFrom<Domain.Entities.Customer>
    {
        public int CustomerID { get; set; }
        public string Name { get; set; }
        public string PAN { get; set; }
        public string AddressPremises { get; set; }
        public string AdressLine1 { get; set; }
        public string AddressLine2 { get; set; }
        public string City { get; set; }
        public string PinCode { get; set; }
        public string MobileNo { get; set; }
        public string EmailID { get; set; }
        public DateTime DateOfBirth { get; set; }
        public int? StateId { get; set; }
        public bool IsTracesRegistered { get; set; }
        public string TracesPassword { get; set; }
        public bool? AllowForm16B { get; set; }
        public string ISD { get; set; }
        public string AlternateNumber { get; set; }
        public string Sellers { get; set; }
        public bool? IsPanVerified { get; set; }
        public bool? OnlyTDS { get; set; }
        public bool? InvalidPAN { get; set; }
        public bool? IncorrectDOB { get; set; }
        public bool? LessThan50L { get; set; }
        public bool? CustomerOptedOut { get; set; }

        public DateTime? CustomerOptingOutDate { get; set; }
        public string CustomerOptingOutRemarks { get; set; }
        public DateTime? InvalidPanDate { get; set; }
        public string InvalidPanRemarks { get; set; }
        public string IncomeTaxPassword { get; set; }
        public virtual ICollection<Domain.Entities.CustomerProperty> CustomerProperty { get; set; }

        public void Mapping(Profile profile)
        {
            profile.CreateMap<Domain.Entities.Customer, CustomerDto>();
        }

    }
}
